framework/src/tinyalsa/tinyalsa.c

```diff
@@ -1119,8 +1119,10 @@ int pcm_stop(struct pcm *pcm)
 	struct audio_msg_s msg;
 	unsigned int size;
 	int prio;
-	while (pcm->buf_ptr > 0) {
-		size = mq_receive(pcm->mq, (FAR char *)&msg, sizeof(msg), &prio);
+	struct timespec st_time;
+	while (1) {
```

Comment:
Instead of `while(1)` , `while(pcm->buf_ptr > 0)` should be better here.

Reply:
I leave it to the tinyalsa team to decide on it.